lib/broadway/topology/rate_limiter.ex

```diff
@@ -79,12 +81,16 @@ defmodule Broadway.Topology.RateLimiter do
     {:reply, %{interval: interval, allowed_messages: allowed}, state}
   end
 
+  def handle_call(:get_rate_limiter_ref, _from, %{counter: counter} = state) do
+    {:reply, counter, state}
+  end
+
   @impl true
   def handle_info(:reset_limit, state) do
-    %{producers_names: producers_names, interval: interval, allowed: allowed, table: table} =
+    %{producers_names: producers_names, interval: interval, allowed: allowed, counter: counter} =
       state
 
-    true = :ets.insert(table, {@row_name, allowed})
+    :atomics.put(counter, 1, allowed)
```

Comment:
With my suggestion above, this would become

```suggestion
    :atomics.put(counter, @atomics_index, allowed)
```